.github/workflows/draft-release.yaml

```diff
@@ -0,0 +1,32 @@
+name: Draft Release
+
+on:
+  push:
+    # Build and publish artifacts when new tag is created for release
+    tags:
+      - "v*.*.*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Build Release Artifacts
+        run: make build-release-artifacts
+
+      - name: Publish Release
```

Comment:
These artifacts will be published along with the release. What should be the workflow for pushing images to Harbor registry? 
- build and push BYOH bundle
- build and push controller image

cc @dharmjit

Reply:
If we also want to push images to the harbor registry then that needs a separate workflow. The other option is to build and push images parallel to building artifacts, in other words, one more action in the current workflow?

```yaml
- name: Publish images to Harbor
  run: make build-publish-images
```

Reply:
Yup. Different workflow makes sense. Let's do it in a separate issue / PR though. Because for draft release, the jobs that you have currently are sufficient.

Reply:
I was thinking what would be the purpose of the draft release? If the idea is to validate the draft release manifests/agent-binary, then we would require a controller image as well.

Reply:
The idea is to automate generating and uploading assets. Otherwise, for each release, we will have to generate manifest, upload assets, and write release notes manually.

This draft release will trigger everything once we create the release tag. We can still update release notes during the draft release. When everything looks great we can publish it.

Reply:
@dharmjit Not sure if a controller image is required because I don't see that in release assets?
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/releases/tag/v0.1.0-alpha.1

Reply:
Controller image and BYOH bundle upload can be a different workflow - discussed [above](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/pull/261#discussion_r754811863)

Pushing to registry can still be manual (for now), first let's validate with alpha.3 if this PR works as expected